### ISSUE TITLE
Microphone audio stream terminates after 10 minutes of recording

### ISSUE THREAD

**Issue by robinmanuelthiel** [Microsoft/cognitive-services-speech-sdk-js#19]
When using `recognizer.startContinuousRecognitionAsync();`, the microphone stream gets terminated after exactly 10:00 minutes. According to the [release notes of version 1.2.0 
](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/releasenotes#speech-service-sdk-110) this issue should have been fixed for the JavaScript SDK and the reconnection gets executed automatically.

My code looks like this:

```javascript
// Create speech and audio config
const speechConfig = SpeechSDK.SpeechConfig.fromEndpoint(new URL('MY_ENDPOINT'), 'MY_KEY');
speechConfig.speechRecognitionLanguage = 'de-DE';
const audioConfig = SpeechSDK.AudioConfig.fromDefaultMicrophoneInput();

// Setup speech recognizer
const recognizer = new SpeechSDK.SpeechRecognizer(this.speechConfig, this.audioConfig);
recognizer.recognized = (s, e) => this.recognized(s, e);

// Start continous recognition
recognizer.startContinuousRecognitionAsync();
```

This works fine for exactly 10:00 minutes talking to the microphone. After that period, I get the following error messages:

![image](https://user-images.githubusercontent.com/5874465/50403846-51cd0d80-07a2-11e9-9989-b687b0d96e55.png)

Am I doing something wrong, or is this not successfully fixed yet?
I am using Angular 6.1 with **version 1.2.0** of the [`microsoft-cognitiveservices-speech-sdk`](https://www.npmjs.com/package/microsoft-cognitiveservices-speech-sdk).

**Comment by rhurey** [comment]
First off, sorry about the response delay, holidays and vacations slowed things down a bit.

Looking at your error something odd is happening here and I'm looking into what the underlying cause is.

**Comment by robinmanuelthiel** [comment]
Quick update: When using the NPM version (I used the downloads form the docs before), the continuous recognition works. After 10 minutes though, I still get the following error in the Debug Console:

![image](https://user-images.githubusercontent.com/5874465/50642007-2c8e8b00-0f6a-11e9-838f-7e271e2fea94.png)

**Comment by jpetitte** [comment]
@robinmanuelthiel I'm seeing this issue now too for some reason. I didn't when I ran some tests when 1.2.0 of the sdk initially came out.

@rhurey Is this related to the fact that access tokens to the speech api are only valid for [10 min](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/rest-apis)?

**Comment by jpetitte** [comment]
@rhurey OK I think I understand. The serverside is closing the connection at 10min, but currently handling that connection disrupt from the library side didn't account for the connection closing during a send operation. 

Does this look like something I can hotfix quickly in the sdk while waiting for a new official version, or should I just manually stop/restart continuous recognition every 9 min (or so)? 

Thanks for looking into this one so fast!

**Comment by rhurey** [comment]
There's a PR for the change. 

The code change itself is pretty small, the test change was larger as I found a couple of other bad behaviors at the same time the previous tests didn't catch.

**Comment by fmegen** [comment]
The pullrequest is in master now.
@robinmanuelthiel please give it a try.

**Comment by robinmanuelthiel** [comment]
Cool, it that something that gets resolved automatically (in the backend) or do I need to update my JS packages? In NPM, I still see version `1.2.0` as the latest one available.

**Comment by robinmanuelthiel** [comment]
Is that something that is open for discussion, @wolfma61? I would argue, that bugfixes like this one should get rolled out as minor updates (0.0.X) as soon as possible. Otherwise, one would have to wait weeks for a bugfix.

I can understand that you want to keep in synch with the other SDKs, but for bugfixes, I see a problem here...

**Comment by robinmanuelthiel** [comment]
Anyways, I gave the new version on `master` try:

**The good news: The reconnection now works**
**The bad news: I still get the error**

After 10:00 minutes, the service now seems to automatically reconnect but in the broser's console, I still see the `WebSocket is already in CLOSING or CLOSED state.` error.

![image](https://user-images.githubusercontent.com/5874465/50924740-bc43a600-1450-11e9-8a1f-1ee7bf0cf601.png)

It also seems like the service would loose some words during the reconnection phase, but I am not 100% sure about that. If that's the case, we should eventually buffer locally.

**Comment by rhurey** [comment]
Well, that is unfortunate you're still seeing the exception in the console. What browser are you using? I'm not seeing it during the reconnect so I'm going to need to poke around more at getting it to fail for me.

Are you seeing speech get lost? There is a buffer already that rewinds the transmitted audio to the end of the last successful recognition and starts to replay at faster than real-time to catch up.

**Comment by rhurey** [comment]
> @rhurey Is this related to the fact that access tokens to the speech api are only valid for [10 min](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/rest-apis)?

Oh, to follow up a bit, you were right and there was some bad error path in the auth token not renewed scenario that was causing the double rejection of a promise error messages. Another PR is out to clean that up.

**Comment by wolfma61** [comment]
@robinmanuelthiel regarding bug fix releases - it is being discussed and I believe there is agreement here on our side. We need to set-up the process for this --- it will probably come soonish

**Comment by robinmanuelthiel** [comment]
Hi @rhurey , as mentioned I am not 100% sure if there were speech losses, I will try to watch closely next time I face the reconnect.

Regarding browsers, I saw that console log in Google Chrome Version 71.0.3578.98 (Official Build) (64-bit) on macOS 10.14.2

**Comment by rhurey** [comment]
Just FYI: you may get a bunch of recognizing events that look like duplicates, but shouldn't loose any audio or get duplicate recognized events.

**Comment by wolfma61** [comment]
sorry for the late reply - we released a 1.2.1 update that has this fi a couple of weeks ago ... 

should have nottified you earlier ... and should have updated the main page ...
 process break-down on my side ...
